src/db/car/many.rs

```diff
@@ -18,19 +18,19 @@ use cid::Cid;
 use fvm_ipld_blockstore::Blockstore;
 use parking_lot::Mutex;
 use positioned_io::RandomAccessFile;
-use std::{fs, io, path::PathBuf, sync::Arc};
+use std::{io, path::PathBuf, sync::Arc};
 
 pub struct ManyCar<WriterT = MemoryDB> {
     shared_cache: Arc<Mutex<ZstdFrameCache>>,
-    read_only: Vec<AnyCar<Box<dyn super::RandomAccessFileReader>>>,
+    read_only: Mutex<Vec<AnyCar<Box<dyn super::RandomAccessFileReader>>>>,
     writer: WriterT,
 }
 
 impl<WriterT> ManyCar<WriterT> {
     pub fn new(writer: WriterT) -> Self {
         ManyCar {
             shared_cache: Arc::new(Mutex::new(ZstdFrameCache::default())),
-            read_only: Vec::new(),
+            read_only: Mutex::new(Vec::new()),
```

Comment:
What's most appropriate here, `Mutex` or `RwLock`?

Reply:
I'm using `Mutex` to align with `shared_cache`. What's your recommendation?

Reply:
Switched to `RwLock`